### ISSUE TITLE
Django 5.0 compatibility

### ISSUE THREAD

**Issue by laymonage** [wagtail/wagtail#9964]
### Issue Summary

[Django 4.2 alpha 1 was released last week](https://www.djangoproject.com/weblog/2023/jan/17/django-42-alpha-1-released/), which means that Django's `main` branch is now on 5.0.

Currently, our `main` branch tests are failing against Django's `main` branch. This issue serves as a tracker of any issues with our code that make it incompatible with Django 5.0.

- [`django.utils.timezone.utc`](https://docs.djangoproject.com/en/4.1/ref/utils/#django.utils.timezone.utc) has been removed. It is an alias to [`datetime.timezone.utc`](https://docs.python.org/3/library/datetime.html#datetime.timezone.utc) and we should use `datetime.timezone.utc` directly. Might need to check if there's any difference to `pytz.utc`, as that seems to be what Django used in earlier versions.
- `assertFormError` and `assertFormsetError` no longer allows passing response and form context name
  - https://github.com/django/django/pull/15179
- The usage of `QuerySet.iterator()` on a queryset that prefetches related objects without providing the `chunk_size` argument is no longer allowed.
- Support for logging out via GET requests in the `django.contrib.auth.views.LogoutView` and `django.contrib.auth.views.logout_then_login()` is removed.

Blocked by https://github.com/wagtail/django-modelcluster/issues/169.

Will update if I find more.

**Comment by laymonage** [comment]
The issue with MySQL seems to affect Django 4.2, too. Haven't checked which commit in Django caused this issue, but the migration that triggers the crash is `wagtailcore.0060_fix_workflow_unique_constraint`, which was introduced in #6607.

This can be replicated through the following steps:

- Install Django 4.2a1 and Wagtail
- Spin up a MySQL server
- Run `python manage.py migrate`, connected to the MySQL server

<details>

<summary>Log</summary>

```
WARNINGS:
wagtailcore.WorkflowState: (models.W036) MySQL does not support unique constraints with conditions.
	HINT: A constraint won't be created. Silence this warning if you don't care about it.
Operations to perform:
  Apply all migrations: admin, auth, base, blog, breads, contenttypes, locations, recipes, sessions, simple_translation, taggit, wagtailadmin, wagtailcore, wagtaildocs, wagtailembeds, wagtailforms, wagtailimages, wagtailredirects, wagtailsearch, wagtailsearchpromotions, wagtailusers
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying auth.0012_alter_user_first_name_max_length... OK
  Applying wagtailcore.0001_squashed_0016_change_page_url_path_to_text_field... OK
  Applying wagtailcore.0017_change_edit_page_permission_description... OK
  Applying wagtailcore.0018_pagerevision_submitted_for_moderation_index... OK
  Applying wagtailcore.0019_verbose_names_cleanup... OK
  Applying wagtailcore.0020_add_index_on_page_first_published_at... OK
  Applying wagtailcore.0021_capitalizeverbose... OK
  Applying wagtailcore.0022_add_site_name... OK
  Applying wagtailcore.0023_alter_page_revision_on_delete_behaviour... OK
  Applying wagtailcore.0024_collection... OK
  Applying wagtailcore.0025_collection_initial_data... OK
  Applying wagtailcore.0026_group_collection_permission... OK
  Applying wagtailcore.0027_fix_collection_path_collation... OK
  Applying wagtailcore.0024_alter_page_content_type_on_delete_behaviour... OK
  Applying wagtailcore.0028_merge... OK
  Applying wagtailcore.0029_unicode_slugfield_dj19... OK
  Applying wagtailcore.0030_index_on_pagerevision_created_at... OK
  Applying wagtailcore.0031_add_page_view_restriction_types... OK
  Applying wagtailcore.0032_add_bulk_delete_page_permission... OK
  Applying wagtailcore.0033_remove_golive_expiry_help_text... OK
  Applying wagtailcore.0034_page_live_revision... OK
  Applying wagtailcore.0035_page_last_published_at... OK
  Applying wagtailcore.0036_populate_page_last_published_at... OK
  Applying wagtailcore.0037_set_page_owner_editable... OK
  Applying wagtailcore.0038_make_first_published_at_editable... OK
  Applying wagtailcore.0039_collectionviewrestriction... OK
  Applying wagtailcore.0040_page_draft_title... OK
  Applying wagtailcore.0041_group_collection_permissions_verbose_name_plural... OK
  Applying wagtailcore.0042_index_on_pagerevision_approved_go_live_at... OK
  Applying wagtailcore.0043_lock_fields... OK
  Applying wagtailcore.0044_add_unlock_grouppagepermission... OK
  Applying wagtailcore.0045_assign_unlock_grouppagepermission... OK
  Applying wagtailcore.0046_site_name_remove_null... OK
  Applying wagtailcore.0047_add_workflow_models... OK
  Applying wagtailcore.0048_add_default_workflows... OK
  Applying wagtailcore.0049_taskstate_finished_by... OK
  Applying wagtailcore.0050_workflow_rejected_to_needs_changes... OK
  Applying wagtailcore.0051_taskstate_comment... OK
  Applying wagtailcore.0052_pagelogentry... OK
  Applying wagtailcore.0053_locale_model... OK
  Applying wagtailcore.0054_initial_locale... OK
  Applying wagtailcore.0055_page_locale_fields... OK
  Applying wagtailcore.0056_page_locale_fields_populate... OK
  Applying wagtailcore.0057_page_locale_fields_notnull... OK
  Applying wagtailcore.0058_page_alias_of... OK
  Applying wagtailcore.0059_apply_collection_ordering... OK
  Applying wagtailcore.0060_fix_workflow_unique_constraint...Traceback (most recent call last):
  File "/Users/sabdullah/Code/github/wagtail/docker-wagtail-develop/wagtail/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/sabdullah/Code/github/wagtail/docker-wagtail-develop/wagtail/venv/lib/python3.10/site-packages/django/db/backends/mysql/base.py", line 75, in execute
    return self.cursor.execute(query, args)
  File "/Users/sabdullah/Code/github/wagtail/docker-wagtail-develop/wagtail/venv/lib/python3.10/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/Users/sabdullah/Code/github/wagtail/docker-wagtail-develop/wagtail/venv/lib/python3.10/site-packages/MySQLdb/cursors.py", line 319, in _query
    db.query(q)
  File "/Users/sabdullah/Code/github/wagtail/docker-wagtail-develop/wagtail/venv/lib/python3.10/site-packages/MySQLdb/connections.py", line 254, in query
    _mysql.connection.query(self, query)
MySQLdb.OperationalError: (1061, "Duplicate key name 'wagtailcore_workflowstate_page_id_6c962862'")
```

</details>

**Comment by laymonage** [comment]
Found it, regression is in https://github.com/django/django/commit/b731e8841558ee4caaba766c83f34ea9c7004f8b introduced in https://github.com/django/django/pull/15254.

**Comment by laymonage** [comment]
Upstream ticket created: https://code.djangoproject.com/ticket/34304

**Comment by laymonage** [comment]
PR opened on upstream: https://github.com/django/django/pull/16509

**Comment by laymonage** [comment]
Upstream PR has been merged, #9974 is now passing.
